### PR TITLE
IPMPROG-5510 Added ssh key document type

### PR DESCRIPTION
**Pull request by sinanyavas-eaton**
- Added ssh key and login document type
- Related unit tests implemented

configuration.json file might need to be changed depending on the use cases of the document.

Signed-off-by: Sinan Yavas [SinanYavas@eaton.com](mailto:SinanYavas@eaton.com)

### CODE REVIEW

File: lib/include/secw_sshkey_and_login.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/*  =========================================================================
+    secw_sshkey_and_login - Document parsers for SshKeyAndLogin document
+    Copyright (C) 2019 - 2023 Eaton
```
Comment (arnaudquette-eaton) [inline lib/include/secw_sshkey_and_login.h]: new implem, (c) should start at the begining year of this implem, so 
Copyright (C) 2023 Eaton

**Review by arnaudquette-eaton — COMMENTED** [review]
I would have liked to see a test with a real SSH key, stored under selftest-ro

File: lib/include/secw_sshkey_and_login.h [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        m_login = login;
+    }
+
+    // Private secw elements
+    const std::string& getSshKey() const
+    {
+        return m_sshKey;
+    }
+    void setSsshKey(const std::string& sshKey)
```
Comment (FrancoisRegisDegott-eaton) [inline lib/include/secw_sshkey_and_login.h]: setSshKey()

File: lib/tests/producer_accessor.cpp [hunk 3]
```diff
@@ -1763,4 +1763,245 @@ void secwProducerAccessorTest(fty::SocketSyncClient& syncClient, mlm::MlmStreamC
...
+            secw::SshKeyAndLoginPtr doc = secw::SshKeyAndLogin::tryToCast(insertedDoc);
+
+            if (doc == nullptr)
+                throw std::runtime_error("No document retrieved");
+
+            if (doc->getUsageIds().count("discovery_monitoring") == 0)
+                throw std::runtime_error("Bad document retrieved: bad usage, discovery_monitoring is missing");
+            if (doc->getUsageIds().size() != 1)
+                throw std::runtime_error("Bad document retrieved: bad usage, bad number of usages id");
+
```
Comment (FrancoisRegisDegott-eaton) [inline lib/tests/producer_accessor.cpp]: rm \n

File: lib/tests/producer_accessor.cpp [hunk 4]
```diff
@@ -1763,4 +1763,245 @@ void secwProducerAccessorTest(fty::SocketSyncClient& syncClient, mlm::MlmStreamC
...
+
+            if (doc->getName() != "Test update sshkey and login")
+                throw std::runtime_error("Bad document retrieved: name do not match");
+            if (doc->getLogin() != "new_login")
+                throw std::runtime_error("Bad document retrieved: login do not match");
```
Comment (FrancoisRegisDegott-eaton) [inline lib/tests/producer_accessor.cpp:1847]: test getSshKey() == "new_key"

is it done for TokenAndLogin document UT?

File: lib/tests/producer_accessor.cpp [hunk 5]
```diff
@@ -1763,4 +1763,245 @@ void secwProducerAccessorTest(fty::SocketSyncClient& syncClient, mlm::MlmStreamC
...
+            if (doc->getUsageIds().count("discovery_monitoring") == 0)
+                throw std::runtime_error("Bad document retrieved: bad usage, discovery_monitoring is missing");
+            if (doc->getUsageIds().size() != 1)
+                throw std::runtime_error("Bad document retrieved: bad usage, bad number of usages id");
+
+
+            if (doc->getId() != id)
+                throw std::runtime_error("Bad document retrieved: id do not match");
+            if (doc->getLogin() != "new_login")
+                throw std::runtime_error("Bad document retrieved: login do not match");
```
Comment (FrancoisRegisDegott-eaton) [inline lib/tests/producer_accessor.cpp:1873]: if (getSshkey()  != "new_key") throw(;;;)

File: lib/tests/producer_accessor.cpp (hunk 4 above)
Comment (sinanyavas-eaton) [reply lib/tests/producer_accessor.cpp:1847]: Yes it's same for the TokenAndLogin doc. But these tests are belong to ProducerAccessor which doesn't have acces to private data so getSshKey() will return empty for all type of docs if the accessor type is producer.

File: lib/tests/producer_accessor.cpp (hunk 5 above)
Comment (sinanyavas-eaton) [reply lib/tests/producer_accessor.cpp:1873]: since it's a ProducerAccessor, it's not expected getSshKey() to return any value.

Comment (FrancoisRegisDegott-eaton) [reply lib/tests/producer_accessor.cpp:1873]: so test: if (getSshkey() != "") throw(;;;)
is it the same for any kind of document? The idea is to test that simple/public accessor can't access private datas...

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

**Pull request merged**